Index: qt-6.10.0/advanced_pdf.cpp
===================================================================
--- qt-6.10.0.orig/advanced_pdf.cpp
+++ qt-6.10.0/advanced_pdf.cpp
@@ -11,6 +11,7 @@
  *   - Adjustment of include statements
  *   - Removal of Q_XXX_EXPORT
  * - Change of the PDF Producer property
+ * - Use of DeviceCMYK color space in PDF output
  */
 // Copyright (C) 2016 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
@@ -1457,7 +1458,7 @@ int AdvancedPdfEngine::metric(QPaintDevi
 AdvancedPdfEnginePrivate::AdvancedPdfEnginePrivate()
     : clipEnabled(false), allClipped(false), hasPen(true), hasBrush(false), simplePen(false),
       needsTransform(false), pdfVersion(AdvancedPdfEngine::Version_1_4),
-      colorModel(AdvancedPdfEngine::ColorModel::Auto),
+      colorModel(AdvancedPdfEngine::ColorModel::CMYK),
       outDevice(nullptr), ownsDevice(false),
       embedFonts(true),
       m_pageLayout(QPageSize(QPageSize::A4), QPageLayout::Portrait, QMarginsF(10, 10, 10, 10))
