--- qt-6.2.0/advanced_pdf.cpp.orig
+++ qt-6.2.0/advanced_pdf.cpp
@@ -22,6 +22,7 @@
  *   - Adjustment of include statements
  *   - Removal of Q_XXX_EXPORT
  *   - Change of the PDF Producer property
+ *   - Use of DeviceCMYK color space in PDF output
  */
 /****************************************************************************
 **
@@ -1261,12 +1262,13 @@ void AdvancedPdfEngine::setPen()
 
     QColor rgba = b.color();
     if (d->grayscale) {
-        qreal gray = qGray(rgba.rgba())/255.;
-        *d->currentPage << gray << gray << gray;
+        qreal gray = (255-qGray(rgba.rgba()))/255.0;
+        *d->currentPage << 0.0 << 0.0 << 0.0 << gray;
     } else {
-        *d->currentPage << rgba.redF()
-                        << rgba.greenF()
-                        << rgba.blueF();
+        *d->currentPage << rgba.cyanF()
+                        << rgba.magentaF()
+                        << rgba.yellowF()
+                        << rgba.blackF();
     }
     *d->currentPage << "SCN\n";
 
@@ -1327,12 +1329,13 @@ void AdvancedPdfEngine::setBrush()
     if (specifyColor) {
         QColor rgba = d->brush.color();
         if (d->grayscale) {
-            qreal gray = qGray(rgba.rgba())/255.;
-            *d->currentPage << gray << gray << gray;
+            qreal gray = (255-qGray(rgba.rgba()))/255.0;
+            *d->currentPage << 0.0 << 0.0 << 0.0 << gray;
         } else {
-            *d->currentPage << rgba.redF()
-                            << rgba.greenF()
-                            << rgba.blueF();
+            *d->currentPage << rgba.cyanF()
+                            << rgba.magentaF()
+                            << rgba.yellowF()
+                            << rgba.blackF();
         }
     }
     if (patternObject)
@@ -1660,7 +1663,7 @@ void AdvancedPdfEnginePrivate::writeHead
 
     // color space for pattern
     patternColorSpace = addXrefEntry(-1);
-    xprintf("[/Pattern /DeviceRGB]\n"
+    xprintf("[/Pattern /DeviceCMYK]\n"
             "endobj\n");
 }
 
@@ -2060,7 +2063,7 @@ void AdvancedPdfEnginePrivate::writePage
     xprintf("<<\n"
             "/ColorSpace <<\n"
             "/PCSp %d 0 R\n"
-            "/CSp /DeviceRGB\n"
+            "/CSp /DeviceCMYK\n"
             "/CSpg /DeviceGray\n"
             ">>\n"
             "/ExtGState <<\n"
