Index: qt-6.7.0/advanced_pdf.cpp
===================================================================
--- qt-6.7.0.orig/advanced_pdf.cpp
+++ qt-6.7.0/advanced_pdf.cpp
@@ -11,6 +11,7 @@
  *   - Removal of Q_XXX_EXPORT
  * - Change of the PDF Producer property
  * - Use of DeviceCMYK color space in PDF output
+ * - Distinct paint engine type
  */
 // Copyright (C) 2016 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
@@ -1356,9 +1357,11 @@ bool AdvancedPdfEngine::newPage()
     return true;
 }
 
+AdvancedPdfEngine::PaintEngineTypeStruct AdvancedPdfEngine::PaintEngineType = {};
+
 QPaintEngine::Type AdvancedPdfEngine::type() const
 {
-    return QPaintEngine::Pdf;
+    return PaintEngineType;
 }
 
 void AdvancedPdfEngine::setResolution(int resolution)
Index: qt-6.7.0/advanced_pdf_p.h
===================================================================
--- qt-6.7.0.orig/advanced_pdf_p.h
+++ qt-6.7.0/advanced_pdf_p.h
@@ -9,6 +9,7 @@
  *   - Renaming of selected files, classes, members and macros
  *   - Adjustment of include statements
  *   - Removal of Q_XXX_EXPORT
+ *   - Distinct paint engine type
  */
 // Copyright (C) 2016 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
@@ -145,6 +146,13 @@ class AdvancedPdfEngine : public QPaintE
     Q_DECLARE_PRIVATE(AdvancedPdfEngine)
     friend class AdvancedPdfWriter;
 public:
+    static struct PaintEngineTypeStruct
+    {
+        constexpr operator QPaintEngine::Type() const {
+            return QPaintEngine::Type(QPaintEngine::User + 1);
+        }
+    } PaintEngineType;
+
     // keep in sync with QPagedPaintDevice::PdfVersion and AdvancedPdfEnginePrivate::writeHeader()::mapping!
     enum PdfVersion
     {
