--- qt-6.2.0/advanced_pdf.cpp.orig
+++ qt-6.2.0/advanced_pdf.cpp
@@ -23,6 +23,7 @@
  *   - Removal of Q_XXX_EXPORT
  *   - Change of the PDF Producer property
  *   - Use of DeviceCMYK color space in PDF output
+ *   - Distinct paint engine type
  */
 /****************************************************************************
 **
@@ -1363,9 +1364,11 @@ bool AdvancedPdfEngine::newPage()
     return true;
 }
 
+AdvancedPdfEngine::PaintEngineTypeStruct AdvancedPdfEngine::PaintEngineType = {};
+
 QPaintEngine::Type AdvancedPdfEngine::type() const
 {
-    return QPaintEngine::Pdf;
+    return PaintEngineType;
 }
 
 void AdvancedPdfEngine::setResolution(int resolution)
--- qt-6.2.0/advanced_pdf_p.h.orig
+++ qt-6.2.0/advanced_pdf_p.h
@@ -21,6 +21,7 @@
  *   - Renaming of selected files, classes, members and macros
  *   - Adjustment of include statements
  *   - Removal of Q_XXX_EXPORT
+ *   - Distinct paint engine type
  */
 /****************************************************************************
 **
@@ -169,6 +170,13 @@ public:
         Version_1_6
     };
 
+    static struct PaintEngineTypeStruct
+    {
+        constexpr operator QPaintEngine::Type() const {
+            return QPaintEngine::Type(QPaintEngine::User + 1);
+        }
+    } PaintEngineType;
+
     AdvancedPdfEngine();
     AdvancedPdfEngine(AdvancedPdfEnginePrivate &d);
     ~AdvancedPdfEngine() {}
