Index: qt-6.10.0/advanced_pdf.cpp
===================================================================
--- qt-6.10.0.orig/advanced_pdf.cpp
+++ qt-6.10.0/advanced_pdf.cpp
@@ -10,6 +10,7 @@
  *   - Renaming of selected files, classes, members and macros
  *   - Adjustment of include statements
  *   - Removal of Q_XXX_EXPORT
+ * - Change of the PDF Producer property
  */
 // Copyright (C) 2016 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
@@ -1758,7 +1759,7 @@ void AdvancedPdfEnginePrivate::writeInfo
     write("\n/Author ");
     printString(author);
     write("\n/Producer ");
-    printString(QString::fromLatin1("Qt " QT_VERSION_STR));
+    printString(QString::fromLatin1("LibreMapper"));
 
     const QTime t = date.time();
     const QDate d = date.date();
@@ -1810,7 +1811,7 @@ int AdvancedPdfEnginePrivate::writeXmpDo
     if (!xmpDocumentMetadata.isEmpty()) {
         metaDataContent = xmpDocumentMetadata;
     } else {
-        const QString producer(QString::fromLatin1("Qt " QT_VERSION_STR));
+        const QString producer(QString::fromLatin1("LibreMapper"));
         const QString metaDataDate = date.toString(Qt::ISODate);
 
         using namespace Qt::Literals;
