--- qt-6.2.0/advanced_pdf.cpp.orig
+++ qt-6.2.0/advanced_pdf.cpp
@@ -21,6 +21,7 @@
  *   - Renaming of selected files, classes, members and macros
  *   - Adjustment of include statements
  *   - Removal of Q_XXX_EXPORT
+ *   - Change of the PDF Producer property
  */
 /****************************************************************************
 **
@@ -1671,7 +1672,7 @@ void AdvancedPdfEnginePrivate::writeInfo
     xprintf("\n/Creator ");
     printString(creator);
     xprintf("\n/Producer ");
-    printString(QString::fromLatin1("Qt " QT_VERSION_STR));
+    printString(QString::fromLatin1("OpenOrienteering"));
     QDateTime now = QDateTime::currentDateTime();
     QTime t = now.time();
     QDate d = now.date();
